### DISCUSSION TITLE
How to use Arg(name='xxx')？

### DISCUSSION THREAD

**Discussion by wu-clan** [DanCardin/cappa#45] [Q&A] [answered]
When I use Arg(name='xxx') to define the parameter name, it always prompts me: `ValueError: Arg 'name' cannot be set when using automatic inference.`

I think name replaces the default uppercase arguments name

**Comment by wu-clan** [comment]
Maybe it's still a little different than I thought, I'm trying to achieve something like the metavar parameter in Typer by modifying name to the input type

**Comment by wu-clan** [comment]
Effect Comparison：

typer define the input type using typer.Option(metavar='xxx')

![image](https://github.com/DanCardin/cappa/assets/52145145/590ea471-afb9-4424-bdc2-5701d742211f)

cappa contains uppercase arguments without input types

![image](https://github.com/DanCardin/cappa/assets/52145145/47975c2e-b88d-4494-9bff-b09aa7deeb43)

**Reply by DanCardin** [reply]
Created an issue for this: https://github.com/DanCardin/cappa/issues/46, tl;dr missing feature currently, should be straightforward to add

**Reply by DanCardin** [reply, marked as answer]
So `Arg("foo")` or `Arg(value_name="foo")` should work as of v0.10.0
